src/components/ModularTable/ModularTable.stories.mdx

```diff
@@ -0,0 +1,68 @@
+import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
+import ModularTable from "./ModularTable";
+
+<Meta title="ModularTable" component={ModularTable} />
+
+### ModularTable
+
+This is a [React](https://reactjs.org/) component to support many table use cases.
+
+### Props
+
+<Props of={ModularTable} />
+
+### Default
+
+<Preview>
+  <Story name="Default">
+    <ModularTable
+      columns={React.useMemo(
```

Comment:
Is the `useMemo` strictly necessary when working with the `react-table`?

- If it is, then I'd prefer it to be abstracted away into the table so we can just provide it with data from immer/redux.
- If it isn't then can you remove it :D

Reply:
It is necessary. I have no idea how they can detect it or why it has any difference (other than simply performance of re-renders), but react-table will not render if data is not memoized.
They clearly state in the API docs which params need to be memoized (https://react-table.tanstack.com/docs/api/useTable).

But I'm not sure if this should be abstracted and done internally by the component itself. Component will not know how this data is build and when should it be refreshed. I'm also not sure what would happen if objects are memoized twice - once by parent component, and second time 'abstracted' inside the table.

Reply:
Just to be sure I verified again, and for some reason now I can't make it fail if I don't memoize the data or columns.
So it seems that not memoizing may not 'break' react-table.
But still, their docs clearly state these arguments as requiring memoization as without it on each re-render table would recalculate a lot of logic. So I think we should keep memoization as requirement.

But I still think memoization should be done wherever the data is prepared for the table (because this component will know for example when to recalculate it (second param to `useMemo` hook)).

Reply:
Ahhh ok, It probably uses a ==='s internally to check if it's the same pointer.

Because of this requirement I think that we need to be very clear in our documentation that it's based off of `react-table` and that this value needs to be memoized as that's not a very common problem (especially if it doesn't output an error in the console.)